yarl/__init__.py

```diff
@@ -557,16 +557,20 @@ def with_port(self, port):
                                                                port)),
                    encoded=True)
 
-    def with_query(self, query):
+    def with_query(self, query=None, **kwargs):
```

Comment:
This prevents from using query as key: `url.with_query(query='val1', subquery='val2')`.
Please change signature to `def with_query(self, *args, **kwargs):` and check for `args` and `kwargs` mutual exclusiveness.
`len(args)` should be 1 if any.